addon/-private/syntax-mount-ext.js

```diff
@@ -0,0 +1,124 @@
+import Ember from 'ember';
+import emberRequire from './ext-require';
+
+const {
+  assert,
+  runInDebug
+} = Ember;
+
+const mountSyntaxModule = Ember.__loader.require('ember-glimmer/syntax/mount');
+const ComponentDefinition = emberRequire('@glimmer/runtime', 'ComponentDefinition');
+const UNDEFINED_REFERENCE = emberRequire('@glimmer/reference', 'UNDEFINED_REFERENCE');
+const RootReference = emberRequire('ember-glimmer/utils/references', 'RootReference');
+const generateControllerFactory = emberRequire('ember-routing', 'generateControllerFactory');
+const OutletLayoutCompiler = emberRequire('ember-glimmer/syntax/outlet', 'OutletLayoutCompiler');
+const FACTORY_FOR = emberRequire('container', 'FACTORY_FOR');
+const AbstractManager = emberRequire('ember-glimmer/syntax/abstract-manager');
+
+function dynamicEngineFor(vm, symbolTable) {
+  let env     = vm.env;
+  let args    = vm.getArgs();
+  let nameRef = args.positional.at(0);
+
+  return new DynamicEngineReference({ nameRef, env, symbolTable });
+}
+
+mountSyntaxModule.mountMacro = function(path, params, hash, builder) {
+  assert(
+    'You can only pass a single argument to the {{mount}} helper, e.g. {{mount "chat-engine"}}.',
+    params.length === 1 && hash === null
+  );
+
+  assert(
+    'The first argument of {{mount}} must be quoted, e.g. {{mount "chat-engine"}}.',
+    typeof params[0] === 'string'
+  );
+
+  let definitionArgs = [params.slice(0, 1), null, null, null];
+  let args = [null, null, null, null];
+  builder.component.dynamic(definitionArgs, dynamicEngineFor, args, builder.symbolTable);
+  return true;
+};
+
+class DynamicEngineReference {
+  constructor({ nameRef, env, symbolTable, args }) {
+    this.tag = nameRef.tag;
+    this.nameRef = nameRef;
+    this.env = env;
+    this.symbolTable = symbolTable;
+    this.args = args;
+  }
+
+  value() {
+    let { env, nameRef, /*symbolTable*/ } = this;
+    let nameOrDef = nameRef.value();
+
+    assert(
+      `You used \`{{mount '${nameOrDef}'}}\`, but the engine '${nameOrDef}' can not be found.`,
+      env.owner.hasRegistration(`engine:${nameOrDef}`)
+    );
+
+    return new MountDefinition(nameOrDef, env);
+  }
```

Comment:
We need to cache the definition here. If the returned value does not `===` the previously returned value, it will cause the component (the `{{mount`) to be torn down.